pkg/login/social/azuread_oauth.go

```diff
@@ -0,0 +1,116 @@
+package social
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/models"
+
+	"golang.org/x/oauth2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+type SocialAzureAD struct {
+	*SocialBase
+	allowedDomains []string
+	allowSignup    bool
+}
+
+type azureClaims struct {
+	Email      string   `json:"email"`
+	UniqueName string   `json:"unique_name"`
```

Comment:
`unique_name` as mentioned in [Azure AD docs](https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#payload-claims) is only issued in v1.0 ID tokens. I think it makes sense to use `preferred_username` intead as a backup field for getting email.

Reply:
You are right. I tested with v1 and v2. I will change this to `preferred_username`. Are you not getting the email even when `email` is added to the scopes? I thought that is standard for openid connect.

Reply:
I didn't get email even if it's added in scopes. But I think it's related more to my account, not to grafana. I use google account for signing in, so maybe email is not set properly. I'll try to add email and check it again.